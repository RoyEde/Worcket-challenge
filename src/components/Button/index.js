import React from 'react';
import { bool } from 'prop-types';

const Button = ({ children, danger, disabled, primary, ...props }) => {
  return (
    <button
      {...props}
      className={`button ${
        primary ? 'button--primary' : ''
      } ${
        disabled ? 'button--disabled' : ''
      } ${
        danger ? 'button--danger' : ''
      }`}
      disabled={disabled}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  danger: bool,
  disabled: bool,
  primary: bool
};

export default Button;
