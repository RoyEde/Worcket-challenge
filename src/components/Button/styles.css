.button {
  appearance: none;
  background-color: #fff;
  border: 2px solid #999999;
  border-radius: 5px;
  cursor: pointer;
  font-family: inherit;
  font-size: 1.2rem;
  font-weight: 600;
  min-width: 40%;
  padding: 0.5rem 1.5rem;
  transition: background-color 0.4s, border-color 0.4s;
}

.button:not(.button--disabled):hover,
.button:not(.button--disabled):focus {
  background-color: var(--button--hover);
}

.button--primary {
  background-color: var(--button--primary-background-color);
  border-color: var(--button--primary-border-color);
  color: var(--button--primary-text-color);
}

.button--disabled {
  background-color: var(--button--disabled-background-color);
  border-color: var(--button--disabled-border-color);
  color: var(--button--disabled-text-color);
  cursor: auto;
}

.button--danger {
  background-color: var(--button--danger-background-color);
  border-color: var(--button--danger-border-color);
  color: var(--button--danger-text-color);
}

.button--primary:not(.button--disabled):hover,
.button--primary:not(.button--disabled):focus {
  background-color: var(--button--primary--hover);
  border-color: var(--button--primary--hover);
}

.button--danger:not(.button--disabled):hover,
.button--danger:not(.button--disabled):focus {
  background-color: var(--button--danger--hover);
  border-color: var(--button--danger--hover);
}
