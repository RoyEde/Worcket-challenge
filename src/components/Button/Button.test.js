import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Button from './index';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshots match', () => {
  const component = renderer.create(<Button />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  tree.props.primary = true;
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  tree.props.disabled = true;
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  tree.props.disabled = true;
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
