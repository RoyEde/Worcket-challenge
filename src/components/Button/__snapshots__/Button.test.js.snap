// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots match 1`] = `
<button
  className="button   "
/>
`;

exports[`snapshots match 2`] = `
<button
  className="button   "
/>
`;

exports[`snapshots match 3`] = `
<button
  className="button   "
/>
`;

exports[`snapshots match 4`] = `
<button
  className="button   "
/>
`;
